examples/examples_yaml_test.go

```diff
@@ -54,8 +58,13 @@ func TestSecretsYAML(t *testing.T) {
 			imgName, ok := stack.Outputs["imageName"]
 			assert.True(t, ok)
 			assert.NotEmpty(t, imgName)
-			assert.Equal(t, "pulumibot/test-secrets:yaml", imgName)
 
+			// imgName is going to be a secret value encoded as a map. Currently ProgramTest lacks the
+			// capacity to decrypt it and check the contents. For now we simply ensure that the secret
+			// information is not present plaintext in the JSON expansion of this value.
+			imgNameStr, err := json.Marshal(imgName)
```

Comment:
OK so this is interesting. This tests was asserting a behavior we do not want actually, buggy behavior that incidentally is getting fixed with this change. The secret data was passed into the Image resource but got plain-texted back, which is a problem. Before this change Docker provider would un-secret all program-originating secret data that got passed through. After the change it's correctly stored as secret. The engine takes care of that after we signal "acceptSecrets: false".

```
name: dockerfile-secret
runtime: yaml
variables:
  secret:
    fn::secret:
      SECRET_ARG: supersecret
  secretName:
    fn::secret: pulumibot/test-secrets:yaml
resources:
  demo-image:
    type: docker:Image
    properties:
      imageName: ${secretName}
      skipPush: true
      build:
        args:
          ${secret}
    options:
      version: v4.1.0
outputs:
  imageName: ${demo-image.imageName}
```